lading/src/signals.rs

```diff
@@ -89,13 +100,50 @@ impl Phase {
         info!(permits = fill, "signaling phase entered");
         self.sem.add_permits(fill);
     }
+
+    /// Wait for any registered peers before returning.
+    #[tracing::instrument]
+    pub async fn wait_for_peers(self) {
+        let peers = self.peers.load(Ordering::Acquire);
+        let _ = self.ack_sem.acquire_many(peers).await;
+    }
+
+    /// Register with the `Phase` owner to avoid the call to `signal` from
```

Comment:
```suggestion
    /// Register with the `Phase` creator to avoid the call to `signal` from
```
?

not sure, the word "owner" just feels a bit nebulous here given the `clone` idiom.